docs/en/reference/upgrading.rst

```diff
@@ -0,0 +1,53 @@
+Upgrading
+=========
+
+New versions of Doctrine come with an upgrade guide named UPGRADE.md.
+This guide documents BC-breaks and deprecations.
+
+Deprecations
+------------
+
+Deprecations are signaled by emitting a silenced ``E_USER_DEPRECATED``
+error, like this:
+
+.. code-block:: php
+
+    <?php
+
+    @trigger_error(
+        'QuantumDefraculator::__invoke() is deprecated.',
+        E_USER_DEPRECATED
+    );
+
+Since this error is silenced, it will not produce any effect unless you
+opt-in by setting up an error handler designed to ignore the silence
+operator in that case. Such an error handler could look like this:
+
+.. code-block:: php
+
+    <?php
+    set_error_handler(function (
```

Comment:
Is it actually a good idea to overwrite handler for all messages like that? Can we maybe suggest this instead only for deprecations?
```php
set_error_handler(
    function (int $code, string $message, string $file, int $line) : void {
        echo sprintf('Deprecation: "%s" in %s:%d', $message, $file, $line);
    },
    E_USER_DEPRECATED
);
```
https://3v4l.org/AFfPk

Reply:
Fixed